src/business_logic/bus_stop/include/Route.hpp

```diff
@@ -12,100 +12,16 @@ namespace cpp_playground::pg_business_logic::bus_stop
         std::vector<std::shared_ptr<Ibus_stop>> stops_list;
     public:
 
-        Route()
-        {
-            stops_list.push_back(std::make_shared<Tterminal>("Bus terminal - 0"));
-            stops_list.push_back(std::make_shared<Tstop>("Bus stop - 1"));
-            stops_list.push_back(std::make_shared<Tstop>("Bus stop - 2"));
-            stops_list.push_back(std::make_shared<Tterminal>("Bus terminal - 3"));
-        }
-
-        int startTerminalBusCount()
-        {
-            return stops_list[0]->counterBussesStop();
-        }
-
-        void vehicleCheckout(Ivehicle* vehicle)
-        {
-            stops_list[0]->addVehicle(vehicle);
-        }
-
-        int routeSize()
-        {
-            return stops_list.size();
-        }
-
-        bool isNextStopBeyondFinal(std::shared_ptr<Ibus_stop> current_position)
-        {
-            auto it = stops_list.begin();
-            for(;it != stops_list.end(); it++)
-            {
-                if(*it == current_position) break;
-            }
-
-            auto starting_terminal_pointer = stops_list.begin();
-
-            if((starting_terminal_pointer - (it + 1)) <= -(static_cast<std::ptrdiff_t>(stops_list.size())))
-            {
-                return true;
-            }
-            else
-            {
-                return false;
-            }
-        }
-
-        bool isNextStopBeyondStarting(std::shared_ptr<Ibus_stop> current_position)
-        {
-            auto it = stops_list.begin();
-            for(;it != stops_list.end(); it++)
-            {
-                if(*it == current_position) break;
-            }
-
-            auto starting_terminal_pointer = stops_list.begin();
-
-            if(starting_terminal_pointer - (it - 1) > 0)
-            {
-                return true;
-            }
-            else
-            {
-                return false;
-            }
-        }
-
-        std::shared_ptr<Ibus_stop> nextStop(std::shared_ptr<Ibus_stop> current_position,int multiplier)
-        {
-            auto it = stops_list.begin();
-            for(;it != stops_list.end(); it++)
-            {
-                if(*it == current_position) break;
-            }
-
-            it += multiplier;
-            return *it;
-        }
-
-        std::shared_ptr<Ibus_stop> previousStop(std::shared_ptr<Ibus_stop> current_position,int multiplier)
-        {
-            auto it = stops_list.begin();
-            for(;it != stops_list.end(); it++)
-            {
-                if(*it == current_position) break;
-            }
-
-            it -= multiplier;
-            return *it;
-        }
-
-        std::shared_ptr<Ibus_stop> getTerminalAddress()
-        {
-            return stops_list[0];
-        }
-
-
-
+        Route();
+
+        int busCount(int stop_number);
+        void vehicleCheckout(Ivehicle* vehicle);
+        int routeSize();
```

Comment:
Consider verbs as more meaningful name for methods, e.g. getStopsCount()

```suggestion
        int getStopsCount();
```